Add LegacyServiceAccountTokenCleanUp feature in beta (#43563)
kubernetes/website: content/en/docs/reference/labels-annotations-taints/_index.md

Used on: Pod

> When a StatefulSet controller creates a Pod for the StatefulSet, it sets this label on that Pod.

Reviewer: If you can, avoid making whitespace changes here (and squash down). It helps avoid merge conflicts.

Author: Revised.

If a legacy token was last used before the cluster gained the feature (added in Kubernetes v1.26),
then the label isn't set.

### kubernetes.io/legacy-token-invalid-since

Type: Label

Example: `kubernetes.io/legacy-token-invalid-since: 2023-10-27`

Used on: Secret

> The control plane only adds this label to auto-generated Secrets that have the type
> `kubernetes.io/service-account-token`.

Reviewer: Is this correct?
```suggestion
The control plane automatically adds this label to auto-generated Secrets that have the type
`kubernetes.io/service-account-token`, provided that you have the
`LegacyServiceAccountTokenCleanUp` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/) enabled.
Kubernetes {{< skew currentVersion >}} enables that behavior by default.
```

Author: Revised.

The value of this label records the date (ISO 8601 format, UTC time zone) when the control plane

> detects that the auto-generated Secret is not used in a specified time (default to be one year).

Reviewer 2: ```suggestion
detects that the auto-generated Secret has not been used for a specified duration (default to be one year).
```

Author: Revised.

> The label starts to be used in Kubernetes v1.29.

Reviewer 2: ```suggestion
The label is introduced in Kubernetes v1.29.
```

Reviewer: Even better:
```diff
-The label is introduced in Kubernetes v1.29.
+This label was introduced in Kubernetes v1.29.
```
The v1.29 docs won't be published until (shortly) after the v1.29 release has happened.

Author: Revised.

If a legacy token was last used before the cluster gained the feature (added in Kubernetes v1.26),
then the label isn't set.

### kubernetes.io/legacy-token-invalid-since

Type: Label

Example: `kubernetes.io/legacy-token-invalid-since: 2023-10-27`

Used on: Secret

> 

Reviewer 3: describe what the label *does* as well (makes the token contained in the Secret invalid for authentication)
